docs/developer/tutorials/snmp/profile-format.md

```diff
@@ -0,0 +1,204 @@
+
+#### Symbol metrics
+
+In profiles, OIDs can be specified as entries containing the `symbol` field:
+
+```yaml
+metrics:
+  - MIB: TCP-MIB
+    symbol:
+      OID: 1.3.6.1.2.1.6.5
+      name: tcpActiveOpens
+```
+
+#### Table metrics
+
+An SNMP table is an object that is composed of multiple entries.
+
+In a MIB file, tables be recognized by the following syntax:
+
+```asn
+exampleTable OBJECT-TYPE
+    SYNTAX   SEQUENCE OF exampleEntry
+    -- ...
+
+exampleEntry OBJECT-TYPE
+   -- ...
+   ::= { exampleTable 1 }
+```
+
+In profiles, tables can be specified as entries containing the `table` and `symbols` fields:
+
+```yaml
+metrics:
+  - MIB: CISCO-PROCESS-MIB
+    table:
+      # Identification of the table which metrics come from.
+      # For example, this device has multiple CPU units; each row
+      # in this table contains information about a single CPU unit.
+      OID: 1.3.6.1.4.1.9.9.109.1.1.1
+      name: cpmCPUTotalTable
+    symbols:
+      # List of symbols ('columns') to retrieve.
+      # Same format as for a single OID.
+      # Each row in the table will emit these metrics.
+      # For example, if we have N CPU units, then the
+      # integration will emit N `snmp.cpmCPUMemoryUsed` metrics.
+      - OID: 1.3.6.1.4.1.9.9.109.1.1.1.1.12
+        name: cpmCPUMemoryUsed
+      # ...
+```
+
+#### Table metrics tagging
+
+It is possible to add tags to metrics retrieved from a table in three ways:
+
+- Using an index:
+
+```yaml
+metrics:
+  - MIB: CISCO-PROCESS-MIB
+    table:
+      OID: 1.3.6.1.4.1.9.9.109.1.1.1
+      name: cpmCPUTotalTable
+    symbols:
+      - OID: 1.3.6.1.4.1.9.9.109.1.1.1.1.12
+        name: cpmCPUMemoryUsed
+    metric_tags:
+      # Add a 'cpu:<row_index>' tag to each metric of each row,
+      # whose value is the index of the row in the table.
+      # This allows querying metrics for a given CPU unit, e.g. 'cpu:1'.
+      - tag: cpu
+        index: 1
+```
+
+- Using a column within the same table:
+
+```yaml
+metrics:
+  - MIB: IF-MIB
+    table:
+      OID: 1.3.6.1.2.1.2.2
+      name: ifTable
+    symbols:
+      - OID: 1.3.6.1.2.1.2.2.1.14
+        name: ifInErrors
+      # ...
+    metric_tags:
+      # Add an 'interface' tag to each metric of each row,
+      # whose value is obtained from the 'ifDescr' column of the row.
+      # This allows querying metrics by interface, e.g. 'interface:eth0'.
+      - tag: interface
+        column:
+          OID: 1.3.6.1.2.1.2.2.1.2
+          name: ifDescr
+```
+
+- Using a column from a different table.
+
+```yaml
+metrics:
+  - MIB: CISCO-IF-EXTENSION-MIB
+    forced_type: monotonic_count
+    table:
+      OID: 1.3.6.1.4.1.9.9.276.1.1.2
+      name: cieIfInterfaceTable
+    symbols:
+      - OID: 1.3.6.1.4.1.9.9.276.1.1.2.1.1
+        name: cieIfResetCount
+    metric_tags:
+      - MIB: IF-MIB
+        column:
+          OID: 1.3.6.1.2.1.31.1.1.1.1
+          name: ifName
+        table: ifXTable
+        tag: interface
+```
+
+### `metric_tags`
```

Comment:
Can we document a guideline for tags? I've seen profiles that try to tag a symbol that is a DisplayString type which is not a great tag value.

Reply:
Added a note about this in a5cfcc74b, let me know how that sounds!